src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,23 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return width.equals(rectangle.width)
+                && length.equals(rectangle.length)
+                && color.equals(rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        return 31 * (width + length) + color.length();
```

Comment:
this hashCode is very simple and may cause lots of collisions. Try to complicate it with hashcodes of Rectangle's fields. and don't forget, that they might be null :)